Code/Framework/AzCore/AzCore/std/ranges/zip_view.h

```diff
@@ -0,0 +1,402 @@
+            {
+                auto r = i;
+                r -= n;
+                return r;
+            }
+            template<bool Enable = conjunction_v<
+                bool_constant<sized_sentinel_for<
+                iterator_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>,
+                iterator_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>>>...
+                >, class = enable_if_t<Enable>>
+                friend constexpr auto operator-(const iterator& x, const iterator& y) -> difference_type
+            {
+                return min_distance_in_views(x, y, AZStd::index_sequence_for<Views...>{});
+            }
+
+            friend constexpr auto iter_move(const iterator& i) noexcept(
+                noexcept((ranges::iter_move(declval<const iterator_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>&>()) && ...)) &&
+                (is_nothrow_move_constructible_v<range_rvalue_reference_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>> && ...))
+            {
+                return ZipViewInternal::tuple_transform(ranges::iter_move, i.m_current);
+            }
+
+            template<bool Enable =
+                conjunction_v<bool_constant<indirectly_swappable<iterator_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>>>...>,
+                class = enable_if_t<Enable>>
+                friend constexpr void iter_swap(const iterator& l, const iterator& r) noexcept(
+                    noexcept(iter_swap_index(l.m_current, r.m_current, AZStd::index_sequence_for<Views...>{})))
+            {
+                iter_swap_with_index(l, r, AZStd::index_sequence_for<Views...>{});
+            }
+
+        private:
+            ZipViewInternal::tuple_or_pair<iterator_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>...> m_current;
+        };
+
+        template<class... Views>
+        template<bool Const>
+        class zip_view<Views...>::sentinel
+            : public enable_if_t<conjunction_v<
+            bool_constant<(sizeof...(Views) > 0)>,
+            bool_constant<input_range<Views>>...,
+            bool_constant<view<Views>>...
+                >, ZipViewInternal::requirements_fulfilled>
+        {
+            friend class zip_view<Views...>;
+            constexpr explicit sentinel(ZipViewInternal::tuple_or_pair<sentinel_t<::AZStd::ranges::Internal::maybe_const<Const, Views>>...> end);
+
+            template<bool OtherConst, size_t... Indices>
+            static constexpr auto min_distance_between_view_iterators(const iterator<OtherConst>& x,
+                const sentinel& y, AZStd::index_sequence<Indices...>) ->
+                common_type_t<range_difference_t<::AZStd::ranges::Internal::maybe_const<OtherConst, Views>>...>;
+
+            template<bool OtherConst>
+            static constexpr auto iterator_tuple_accessor(const iterator<OtherConst>& it)
+            {
+                return it.m_current;
```

Comment:
There's an .inl file for this header so shouldn't all implementations be moved there?

Reply:
Yeah about this one.
This was in the .inl and the friend `operator==` below was angry about it for some reason.
I can try moving back there again.